misc/pytorch_toolkit/breast_ultrasound_simulation/src/dataloader.py

```diff
@@ -0,0 +1,82 @@
+import os
+from PIL import Image
+from torch.utils import data
+from torchvision import transforms
+
+
+class Kaggle_BUS_Dataset(data.Dataset):
+    def __init__(
+            self,
+            file_names,
+            Stage0_DIR,
+            Images_DIR,
+            resize=True,
+            test=0):
+        # Initiliaztion
+        self.file_names = file_names
+        self.Stage0_DIR = Stage0_DIR
+        self.Images_DIR = Images_DIR
+        self.ten_trans = transforms.Compose(
+            [transforms.ToTensor()])
+        self.resize = resize
+        self.test = test
+
+    def __len__(self):
+        # Return total number of samples in data set
+        return len(self.file_names)
+
+    def __getitem__(self, index):
+
+        file_name = self.file_names[index]
+
+        # stage 0 is input,x
+        X = Image.open(os.path.join(self.Stage0_DIR, file_name))
+        X = self.ten_trans(X)
+
+        # images are the output, y
+        y = Image.open(os.path.join(self.Images_DIR, file_name))
+        # if self.resize:
+        #     y = y.resize((128, 128))
+
+        y = self.ten_trans(y)
+
+        return X, y, file_name
+
+
+class BUS_dataset(data.Dataset):
```

Comment:
The class is very similar to the above one. You can create a base class and move common functionality like `__len__` (and `__init__` since they are almost the same if we do not pay attention to the field names) to it.